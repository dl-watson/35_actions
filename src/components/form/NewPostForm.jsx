import React, { useState } from "react";
import { useDispatch } from "react-redux";
import { createPost } from "../../actions/postActions";
import styles from "./styles/NewPostForm.css";

const NewPostForm = () => {
  const dispatch = useDispatch();

  const [title, setTitle] = useState("");
  const [body, setBody] = useState("");

  const handleSubmit = (e) => {
    e.preventDefault();

    dispatch(createPost({ id: Date.now(), title, body, postcomments: [] }));
  };

  return (
    <section className={styles.NewPostForm} onSubmit={handleSubmit}>
      <h2>Create Post</h2>
      <form>
        <input
          name="title"
          value={title}
          placeholder="Title..."
          onChange={({ target }) => setTitle(target.value)}
        />
        <textarea
          name="body"
          value={body}
          placeholder="Body..."
          onChange={({ target }) => setBody(target.value)}
        />
        <button>Submit</button>
      </form>
    </section>
  );
};

export default NewPostForm;
