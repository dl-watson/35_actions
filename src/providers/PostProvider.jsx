import React, { createContext, useContext, useReducer } from "react";
import PropTypes from "prop-types";
import { postReducer, initialState } from "../reducers/postReducer";

const PostContext = createContext(null);

export const PostProvider = ({ children }) => {
  const [state, dispatch] = useReducer(postReducer, initialState);

  return (
    <PostContext.Provider value={{ state, dispatch }}>
      {children}
    </PostContext.Provider>
  );
};

export const useSelector = (selector) => {
  const { state } = useContext(PostContext);

  return selector(state);
};

export const useDispatch = () => {
  const { dispatch } = useContext(PostContext);
  return dispatch;
};

PostProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
