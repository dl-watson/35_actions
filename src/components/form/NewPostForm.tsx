/* eslint-disable no-unused-vars */
import React, { MouseEvent, useState } from "react";
import { useDispatch } from "react-redux";
import { createPost } from "../../actions/postActions";
import * as styles from "./styles/NewPostForm.css";

const NewPostForm = () => {
  const dispatch = useDispatch();

  const [title, setTitle] = useState("");
  const [body, setBody] = useState("");

  const handleSubmit = (event: MouseEvent) => {
    event.preventDefault();

    dispatch(createPost({ id: Date.now(), title, body }));
  };

  return (
    <section className={styles.NewPostForm} onSubmit={(event) => handleSubmit}>
      <h2>Create Post</h2>
      <form>
        <input
          name="title"
          value={title}
          placeholder="Title..."
          onChange={({ target }) => setTitle(target.value)}
        />
        <textarea
          name="body"
          value={body}
          placeholder="Body..."
          onChange={({ target }) => setBody(target.value)}
        />
        <button>Submit</button>
      </form>
    </section>
  );
};

export default NewPostForm;
