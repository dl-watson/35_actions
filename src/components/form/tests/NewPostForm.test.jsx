import React from "react";
import { render, cleanup } from "@testing-library/react";
import NewPostForm from "../NewPostForm";
import { PostProvider } from "../../../providers/PostProvider";

describe("NewPostForm component", () => {
  afterEach(() => cleanup());
  it("renders NewPostForm", () => {
    const { asFragment } = render(
      <PostProvider>
        <NewPostForm />
      </PostProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
