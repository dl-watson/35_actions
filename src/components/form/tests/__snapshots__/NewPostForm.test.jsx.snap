// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPostForm component renders NewPostForm 1`] = `
<DocumentFragment>
  <section
    class="NewPostForm"
  >
    <h2>
      Create Post
    </h2>
    <form>
      <input
        name="title"
        placeholder="Title..."
        value=""
      />
      <textarea
        name="body"
        placeholder="Body..."
      />
      <button>
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
